src/Microsoft.Health.Dicom.Core/Features/ExtendedQueryTag/GetExtendedQueryTagErrorsService.cs

```diff
@@ -0,0 +1,52 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Common;
+using Microsoft.Health.Dicom.Core.Messages.ExtendedQueryTag;
+
+namespace Microsoft.Health.Dicom.Core.Features.ExtendedQueryTag
+{
+    public class GetExtendedQueryTagErrorsService : IGetExtendedQueryTagErrorsService
+    {
+        private readonly IExtendedQueryTagErrorStore _extendedQueryTagErrorStore;
+        private readonly IDicomTagParser _dicomTagParser;
+        public GetExtendedQueryTagErrorsService(IExtendedQueryTagErrorStore extendedQueryTagStore, IDicomTagParser dicomTagParser)
+        {
+            _extendedQueryTagErrorStore = EnsureArg.IsNotNull(extendedQueryTagStore, nameof(extendedQueryTagStore));
+            _dicomTagParser = EnsureArg.IsNotNull(dicomTagParser, nameof(dicomTagParser));
+        }
+
+        public async Task<GetExtendedQueryTagErrorsResponse> GetExtendedQueryTagErrorsAsync(string tagPath, CancellationToken cancellationToken = default)
+        {
+            string numericalTagPath = null;
```

Comment:
nit: I think since C# 9 you can use `throw` in expressions, so I think the following is valid:
```csharp
string numericalTagPath = _dicomTagParser.TryParse(tagPath, out DicomTag[] tags, supportMultiple: false)
    ? tags[0].GetPath()
    : throw new InvalidExtendedQueryTagPathException(string.Format(DicomCoreResource.InvalidExtendedQueryTag, tagPath ?? string.Empty));
```

Reply:
Stylistically you could probably also do something like:
```csharp
if (!_dicomTagParser.TryParse(tagPath, out DicomTag[] tags, supportMultiple: false))
{
    throw new InvalidExtendedQueryTagPathException(string.Format(DicomCoreResource.InvalidExtendedQueryTag, tagPath ?? string.Empty));
}

// Continue ...
```